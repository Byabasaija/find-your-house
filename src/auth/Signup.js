/* eslint-disable camelcase */
import React, { useState } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import registerUserAction from '../actions/registerUser';
// import Login from './Login';
import './Login.css';

const Signup = ({ signUpUser }) => {
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [password_confirmation, setPassword_confirmation] = useState('');

  const handleChangeUsername = (e) => {
    setUsername(e.target.value);
  };

  const handleChangePassword = (e) => {
    setPassword(e.target.value);
  };

  const handleChangePasswordCof = (e) => {
    setPassword_confirmation(e.target.value);
  };

  const handleSubmit = (e) => {
    const user = { username, password, password_confirmation };
    signUpUser(user);
    e.preventDefault();
  };

  return (

    <div className="limiter">

      <div className="container-login100 login-pg">
        <div className="wrap-login100 p-t-190 p-b-30">
          <form onSubmit={(e) => handleSubmit(e)} className="login100-form validate-form">
            <div className="form-group pb-3 wrap-input100 validate-input m-b-10">
              <input
                type="text"
                name="username"
                placeholder="username"
                value={username}
                onChange={(e) => handleChangeUsername(e)}
                required
                className="form-control p-2 input100"
              />
            </div>

            <div className="form-group pb-3 wrap-input100 validate-input m-b-10">
              <input
                type="password"
                name="password"
                placeholder="Password"
                value={password}
                onChange={(e) => handleChangePassword(e)}
                required
                className="form-control p-2 input100"
              />
            </div>

            <div className="form-group pb-3 wrap-input100 validate-input m-b-10">
              <input
                type="password"
                name="password_confirmation"
                placeholder="Password confirmation"
                value={password_confirmation}
                onChange={(e) => handleChangePasswordCof(e)}
                required
                className="form-control p-2 input100"
              />
            </div>

            <div className="container-login100-form-btn p-t-10">
              <button type="submit" className="login100-form-btn">Register</button>
            </div>
          </form>
          <div>
            <p className="text-center w-full">Already have an account? </p>
            <Link to="/" className="text-center w-full">
              {' '}
              <p>Login</p>
            </Link>
          </div>
        </div>
      </div>
    </div>
  );
};

const mapDispatchToProps = (dispatch) => ({
  signUpUser: (user) => {
    dispatch(registerUserAction(user));
  },
});

Signup.propTypes = {
  signUpUser: PropTypes.func,
};

Signup.defaultProps = {
  signUpUser: () => {},
};
export default connect(null, mapDispatchToProps)(Signup);
