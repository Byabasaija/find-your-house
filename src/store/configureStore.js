import { applyMiddleware, createStore, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers/index';

const saveToLocalStorage = (state) => {
  try {
    const serialisedState = JSON.stringify(state);
    localStorage.setItem('persistantState', serialisedState);
  } catch (e) {
    console.warn(e);
  }
};
const loadFromLocalStorage = () => {
  try {
    const serialisedState = localStorage.getItem('persistantState');
    if (serialisedState === null) return undefined;
    return JSON.parse(serialisedState);
  } catch (e) {
    console.warn(e);
    return undefined;
  }
};

const composeEnhancer = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(rootReducer, loadFromLocalStorage(),
  composeEnhancer(applyMiddleware(thunk)));

store.subscribe(() => saveToLocalStorage(store.getState()));

export default store;
