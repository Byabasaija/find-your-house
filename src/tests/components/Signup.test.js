import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import store from '../../store/configureStore';
import Signup from '../../auth/Signup';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Signup />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
