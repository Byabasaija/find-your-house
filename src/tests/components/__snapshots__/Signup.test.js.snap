// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="limiter"
>
  <div
    className="container-login100 login-pg"
  >
    <div
      className="wrap-login100 p-t-190 p-b-30"
    >
      <form
        className="login100-form validate-form"
        onSubmit={[Function]}
      >
        <div
          className="form-group pb-3 wrap-input100 validate-input m-b-10"
        >
          <input
            className="form-control p-2 input100"
            name="username"
            onChange={[Function]}
            placeholder="username"
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group pb-3 wrap-input100 validate-input m-b-10"
        >
          <input
            className="form-control p-2 input100"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group pb-3 wrap-input100 validate-input m-b-10"
        >
          <input
            className="form-control p-2 input100"
            name="password_confirmation"
            onChange={[Function]}
            placeholder="Password confirmation"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="container-login100-form-btn p-t-10"
        >
          <button
            className="login100-form-btn"
            type="submit"
          >
            Register
          </button>
        </div>
      </form>
      <div>
        <p
          className="text-center w-full"
        >
          Already have an account? 
        </p>
        <a
          className="text-center w-full"
          href="/"
          onClick={[Function]}
        >
           
          <p>
            Login
          </p>
        </a>
      </div>
    </div>
  </div>
</div>
`;
